### PR TITLE
Better pretrained weight loading

### PR DESCRIPTION
**Pull request by klemen1999**
Future proofing for pretrained weights so we can add them later one. Once available they will be automatically downloaded and used.

### CODE REVIEW

File: luxonis_train/nodes/backbones/ddrnet/ddrnet.py [hunk 1]
```diff
@@ -241,8 +241,13 @@ def __init__(
...
         self.init_params()
 
-        if download_weights and var.weights_path:
-            self.load_checkpoint(var.weights_path)
+        if download_weights:
+            if var.weights_path:
+                self.load_checkpoint(var.weights_path)
+            else:
+                logger.warning(
+                    f"No checkpoint available for {self.__class__.__name__}, skipping."
```
Comment (kozlov721) [inline luxonis_train/nodes/backbones/ddrnet/ddrnet.py:249]: There's the `BaseNode.name` property that can be used instead of `__class__.__name__`.

**Review by kozlov721 — APPROVED** [review]
LGTM, just left a small comment. We could also add some tests for the various `get_variant_weights` functions that could just ping the URL to verify it's correct.

**Comment by klemen1999** [comment]
> We could also add some tests for the various get_variant_weights functions that could just ping the URL to verify it's correct.

I agree, the only issue currently is that most of these URLs don't exist yet. They will be there when release is made and the checkpoints are added to the release. So I'll add these tests after release is made.

**Comment by kozlov721** [comment]
> So I'll add these tests after release is made.

Sounds good

**Pull request merged**